subprojects/file-collections/src/main/java/org/gradle/api/internal/file/AbstractFileCollection.java

```diff
@@ -55,15 +56,21 @@
 public abstract class AbstractFileCollection implements FileCollectionInternal {
     protected final TaskDependencyFactory taskDependencyFactory;
     protected final Factory<PatternSet> patternSetFactory;
+    protected final FileCollectionListener fileCollectionListener;
 
-    protected AbstractFileCollection(TaskDependencyFactory taskDependencyFactory, Factory<PatternSet> patternSetFactory) {
+    protected AbstractFileCollection(
+        TaskDependencyFactory taskDependencyFactory,
+        Factory<PatternSet> patternSetFactory,
+        FileCollectionListener fileCollectionListener
```

Comment:
I believe that DI is a correct way to pass listener. The downside of this amount of changes that needed to satisfy new dependency